tests/misc/t8545.nim

```diff
@@ -1,5 +1,6 @@
 discard """
-  targets: "c cpp js"
+  # just tests that this doesn't crash the compiler
+  errormsg: "cannot instantiate: 'a:type'"
```

Comment:
This test only compiled before because `a` was never used, if you tried to use it you would get this same error. I have a stashed diff that gets this test working and lets you actually use `a` (except `cast[static[bool]](a)` which makes no sense) but this PR is already crowded.